### PR TITLE
Hooks migration and other updates

### LINKED ISSUE #229: Next milestones
I'd like to open up conversation on what the next milestones for this project could look like. React is moving quickly and I would like to start taking advantage of some of the new features that it offers.

With that said, here are some ideas that I've had floating around:

1. Change peer dependency of React to >= 16.7, this will close #204 
    1.1 Replace all classes with [hooks](https://reactjs.org/docs/hooks-reference.html). I have already done most of this work on a branch and will be submitting that soon-ish.
    1.2 Determine what components, if any, could use the lazy or memo features of React's API.
2. Publish only modules
3. Simplify existing components as much as possible, some possible examples:
      - New button components for various button types - OutlineButton, LinkButton
      - Extract calls to action into their own component
4. Extract common functionality 
    4.1 Validation displays on input types
    4.2 Additional help displays on input types

I would also like to see about doing an accessibility audit of what we have in order to make sure we're meeting WCAG  2.1 AA standards, verifying that components make sense to a screen reader, and keyboard focus is accounted for as necessary. 

A lot of these would require breaking changes, so I would like to try and make these changes all part of the next major release and try to line it up with the final release of hooks in React. 

What are your thoughts on this? What additional items can we add to this?

### PR DESCRIPTION
**Pull request by aabenoja**
Closes #229

Migration guide pending

### CODE REVIEW

File: packages/es-components/src/components/controls/radio-buttons/RadioGroup.js [hunk 1]
```diff
@@ -33,16 +33,12 @@ function RadioGroup({
   onChange,
   introContent,
   validationState,
-  additionalHelpContent,
-  theme
+  additionalHelpContent
 }) {
+  const theme = useTheme();
```
Comment (aabenoja) [inline packages/es-components/src/components/controls/radio-buttons/RadioGroup.js]: `RadioButton` itself is using this `useTheme` hook, so there isn't a need to pass this as a prop.

**Comment by jrios** [comment]
@Darrken @aabenoja I rebased this branch with the changes from the `master` branch. you will want to update your versions accordingly.

**Comment by aabenoja** [comment]
React 16.8 is live with the hooks API. We just need to write a migration guide for this PR.

**Comment by jrios** [comment]
@aabenoja this is on my to do list. I'm hoping to have something ready for y'all early next week so we can release mid-week or so.

File: es-components.code-workspace [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+	"folders": [
+		{
+			"path": "packages/es-components"
+		}
+	],
+	"settings": {}
+}
```
Comment (aabenoja) [inline es-components.code-workspace]: Is this file for `vscode`?

File: packages/es-components/src/components/patterns/datepicker/DatePicker.js [hunk 3]
```diff
@@ -203,12 +205,7 @@ DatePicker.defaultProps = {
...
-  validationState: 'default'
+  allowNativeDatepickerOnMobile: true
 };
 
-const UncontrolledDatePicker = uncontrollable(DatePicker, {
-  selectedDate: 'onChange'
-});
-
-export default withTheme(withWindowSize(UncontrolledDatePicker));
+export default withTheme(withWindowSize(DatePicker));
```
Comment (aabenoja) [inline packages/es-components/src/components/patterns/datepicker/DatePicker.js]: This is mainly a reminder for myself to remove the `withTheme` usage.

@jrios I started looking at a `useWindowWidth` hook similar to [this gist and comments](https://gist.github.com/gaearon/cb5add26336003ed8c0004c4ba820eae). A lot of the tests around `DatePicker` use the `defaultWidth` prop to mock the state. Setting the screen size at the start of each of the affected tests didn't seem to resolve the errors I ran into. What are your thoughts on this?

Comment (jrios) [reply packages/es-components/src/components/patterns/datepicker/DatePicker.js]: 👍 on `useWindowWidth`. 

What errors are you referring to?

Comment (aabenoja) [reply packages/es-components/src/components/patterns/datepicker/DatePicker.js]: I was running into errors around mocking `document.body.clientWidth` in the `DatePicker` tests. I've got it all working now in #295

**Pull request merged**